cognite/neat/data_model/models/data_types.py

```diff
@@ -0,0 +1,392 @@
+import re
+import sys
+import typing
+from datetime import date, datetime
+from typing import Any, ClassVar
+
+from cognite.client.data_classes import data_modeling as dms
+from cognite.client.data_classes.data_modeling.data_types import Enum as DMSEnum
+from pydantic import BaseModel, Field, model_serializer
+
+from cognite.neat.data_model._constants import XML_SCHEMA_NAMESPACE
+from cognite.neat.data_model._identifiers import URI
+from cognite.neat.data_model.models.entities import (
+    ConceptEntity,
+    UnitEntity,
+)
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+# This patterns matches a string that is a data type, with optional content in parentheses.
+# For example, it matches "float(unit=power:megaw)" as name="float" and content="unit=power:megaw"
+_DATATYPE_PATTERN = re.compile(r"^(?P<name>[^(:]*)(\((?P<content>.+)\))?$")
+
+
+class DataType(BaseModel):
```

Comment:
What about making this an Entity? Have a shared Entiy base class for all entities including data types?